tests/robustness/powerfailure_test.go

```diff
@@ -78,13 +78,74 @@ func TestRestartFromPowerFailure(t *testing.T) {
 		},
 	} {
 		t.Run(tc.name, func(t *testing.T) {
-			doPowerFailure(t, tc.du, tc.fsMountOpt, tc.useFailpoint)
+			doPowerFailure(t, tc.du, dmflakey.FSTypeEXT4, "", tc.fsMountOpt, tc.useFailpoint)
 		})
 	}
 }
 
-func doPowerFailure(t *testing.T, du time.Duration, fsMountOpt string, useFailpoint bool) {
-	flakey := initFlakeyDevice(t, strings.Replace(t.Name(), "/", "_", -1), dmflakey.FSTypeEXT4, fsMountOpt)
+func TestRestartFromPowerFailureXFS(t *testing.T) {
+	for _, tc := range []struct {
+		name         string
+		du           time.Duration
+		mkfsOpt      string
+		fsMountOpt   string
+		useFailpoint bool
+	}{
+		{
+			name:         "xfs_no_opts",
+			du:           5 * time.Second,
+			mkfsOpt:      "",
+			fsMountOpt:   "",
+			useFailpoint: true,
+		},
+		{
+			name:         "lazy-log",
+			du:           5 * time.Second,
+			mkfsOpt:      "-l lazy-count=1",
+			fsMountOpt:   "",
+			useFailpoint: true,
+		},
+		{
+			name:         "odd-allocsize",
+			du:           5 * time.Second,
+			mkfsOpt:      "",
+			fsMountOpt:   "allocsize=" + fmt.Sprintf("%d", 4096*5),
+			useFailpoint: true,
+		},
+		{
+			name:         "nolargeio",
+			du:           5 * time.Second,
+			mkfsOpt:      "",
+			fsMountOpt:   "nolargeio",
+			useFailpoint: true,
+		},
+		{
+			name:         "odd-alignment",
+			du:           5 * time.Second,
+			mkfsOpt:      "-d sunit=1024,swidth=1024",
+			fsMountOpt:   "noalign",
+			useFailpoint: true,
+		},
+		{
+			name:    "openshift-sno-options",
+			du:      5 * time.Second,
+			mkfsOpt: "-m bigtime=1,finobt=1,rmapbt=0,reflink=1 -i sparse=1 -l lazy-count=1",
+			// openshift also supplies seclabel,relatime,prjquota on RHEL, but that's not supported on our CI
+			// prjquota is only unsupported on our ARM runners.
+			fsMountOpt:   "rw,attr2,inode64,logbufs=8,logbsize=32k",
```

Comment:
Please provide a link to the guide of XFS, including detailed explanation to all the parameters.

Reply:
I've left a mention of the `man` commands, not sure it makes sense to link documentation to an online manual.